core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -5914,6 +5866,73 @@ enum DefaultConstants NADefaults::validateAndInsert(const char *attrName,
 
 } // NADefaults::validateAndInsert()
 
+float NADefaults::computeNumESPsPerCore(NABoolean aggressive)
+{
+   #define DEFAULT_ESPS_PER_NODE 2   // for conservation allocation
+   #define DEFAULT_ESPS_PER_CORE 0.5 // for aggressive allocation
+
+     // Make sure the gpClusterInfo points at an NAClusterLinux object.
+     // In osim simulation mode, the pointer can point at a NAClusterNSK
+     // object, for which the method numTSEsForPOS() is not defined.
+   NAClusterInfoLinux* gpLinux = dynamic_cast<NAClusterInfoLinux*>(gpClusterInfo);
+   assert(gpLinux);		
+
+   // cores per node
+   Lng32 coresPerNode = gpClusterInfo->numberOfCpusPerSMP();
+
+   if ( aggressive ) {
+      float totalMemory = gpLinux->totalMemoryAvailable(); // per Node, in KB
+      totalMemory /= 1024; // per Node, in GB
+      totalMemory /= coresPerNode ; // per core, in GB
+      totalMemory /= 2; // per core, 2GB per ESP
+      return MINOF(DEFAULT_ESPS_PER_CORE, totalMemory);
+   }
+
+   Lng32 numESPsPerNode = DEFAULT_ESPS_PER_NODE;
+
+     // number of POS TSE
+   Lng32 numTSEsPerCluster = gpLinux->numTSEsForPOS();
```

Comment:
Probably wait? I am not sure if TSE-style storage is possible in the
future.

On Wed, Jan 6, 2016 at 5:21 PM, Hans Zeller <notifications@github.com>
wrote:

> In core/sql/sqlcomp/nadefaults.cpp
> <https://github.com/apache/incubator-trafodion/pull/250#discussion_r49023989>
> :
>
> > +
> > +   // cores per node
> > +   Lng32 coresPerNode = gpClusterInfo->numberOfCpusPerSMP();
> > +
> > +   if ( aggressive ) {
> > +      float totalMemory = gpLinux->totalMemoryAvailable(); // per Node, in KB
> > +      totalMemory /= 1024; // per Node, in GB
> > +      totalMemory /= coresPerNode ; // per core, in GB
> > +      totalMemory /= 2; // per core, 2GB per ESP
> > +      return MINOF(DEFAULT_ESPS_PER_CORE, totalMemory);
> > +   }
> > +
> > +   Lng32 numESPsPerNode = DEFAULT_ESPS_PER_NODE;
> > +
> > +     // number of POS TSE
> > +   Lng32 numTSEsPerCluster = gpLinux->numTSEsForPOS();
>
> Since there are no TSEs in Trafodion, can we get rid of this code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/250/files#r49023989>.
>



-- 
Regards, --Qifan